### DISCUSSION TITLE
PEPs Worth Supporting

### DISCUSSION THREAD

**Discussion by mitsuhiko** [astral-sh/rye#9] [General]
Here are some PEPs that would make Rye's life easier (or any tool of that nature). Today I believe only one PEP that is not accepted has relevance:

* [PEP 711](https://peps.python.org/pep-0711/) ([discussion thread](https://discuss.python.org/t/pep-711-pybi-a-standard-format-for-distributing-python-binaries/25547)): this PEP proposes binary distributions for unpacking. Today Rye depends on [indygreg's Python builds](https://github.com/indygreg/python-build-standalone). It would be preferable to get such builds directly from CPython itself.

Important PEPs that are in progress by the ecosystem:

* [PEP 658](https://peps.python.org/pep-0658/): for exposing wheel metadata from PyPI. Accepted but [implementation is slow](https://github.com/pypi/warehouse/issues/8254).

**Comment by Zomatree** [comment]
While its not a PEP yet, I've opened a discussion on dev dependencies for the pyproject format, if you have anything you would like to add it would be much appreciated and maybe helps push it along 😃, because I saw on your readme that rye does not support dev dependencies yet due to the fact that there is no standard for it yet.

https://discuss.python.org/t/development-dependencies-in-pyproject-toml/26149

**Reply by mitsuhiko** [reply]
It does support dev-dependencies, but it has to use a made up format :(

**Comment by jeanas** [comment]
https://discuss.python.org/t/pep-658-is-now-live-on-pypi/26693
